### PR TITLE
ORLP-217 fix user status with message page for web

### PR DESCRIPTION
**Pull request by Tamara20**
check user status when user want to rating courses or decks

### CODE REVIEW

**Comment by Tamara20** [comment]
what is better write one annotation and use it with methods which I need, but in this case when annotation is satisfied I take data from DB two times: first in annotation and second in my method Is it good practice. Or write method 
    public void isUserStatusActive(User user) throws UserStatusException {
        if (user.getAccount().getStatus().isNotActive()) {
            throw new UserStatusException(user.getAccount().getStatus());
        }
    }
and remember call this method in each method if it needs but in this case I will take date from data base one time.

**Comment by antonbakalets** [comment]
I thought the user status is available in Spring security context. But now I see that a JwtUser is stored there. In that case your comment about two transactions to database make sense.
You can switch to the implementation without annotation, but please update your branch from master.

**Comment by Tamara20** [comment]
Yes, of course, thank you

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckRatingService.java [hunk 1]
```diff
@@ -1,6 +1,8 @@
 package com.softserve.academy.spaced.repetition.service;
 
+import com.softserve.academy.spaced.repetition.exceptions.UserStatusException;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckRatingService.java]: Sonar will report an issue.

Comment (b0hdan) [reply src/main/java/com/softserve/academy/spaced/repetition/service/DeckRatingService.java]: Why?

Comment (antonbakalets) [reply src/main/java/com/softserve/academy/spaced/repetition/service/DeckRatingService.java]: Please install the plugin and check.

**Pull request merged**